internal/generator/vector/output/azuremonitor/azuremonitor_test.go

```diff
@@ -0,0 +1,140 @@
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	sharedKey     = "z9ndQSFH1RLDnS6WR35m84u326p3"
+	azureId       = "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/otherResourceGroup/providers/Microsoft.Storage/storageAccounts/examplestorage"
+	hostCN        = "ods.opinsights.azure.cn"
+	customerId    = "6vzw6sHc-0bba-6sHc-4b6c-8bz7sr5eggRt"
+	secretName    = "azure-monitor-secret"
+	secretTlsName = "azure-monitor-secret-tls"
+	outputName    = "azure_monitor_logs"
+	logType       = "myLogType"
+)
+
+//go:embed azm_common.toml
+var ExpectedAzureCommonToml string
+
+//go:embed azm_advance.toml
+var ExpectedAzureAdvanceToml string
+
+//go:embed azm_tls.toml
+var ExpectedAzureTlsToml string
+
+//go:embed azm_skip_tls.toml
+var ExpectedAzureSkipTls string
+
+var _ = Describe("Generating vector config for Azure Monitor Logs output:", func() {
+
+	var (
+		g framework.Generator
+
+		outputCommon = loggingv1.OutputSpec{
+			Type: loggingv1.OutputTypeAzureMonitor,
+			Name: outputName,
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				AzureMonitor: &loggingv1.AzureMonitor{
+					CustomerId: customerId,
+					LogType:    logType,
+				},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: secretName,
+			},
+		}
+
+		outputCommonTls = loggingv1.OutputSpec{
+			Type: loggingv1.OutputTypeAzureMonitor,
+			Name: outputName,
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				AzureMonitor: &loggingv1.AzureMonitor{
+					CustomerId: customerId,
+					LogType:    logType,
+				},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: secretTlsName,
+			},
+		}
+
+		outputAdvance = loggingv1.OutputSpec{
+			Type: loggingv1.OutputTypeAzureMonitor,
+			Name: outputName,
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				AzureMonitor: &loggingv1.AzureMonitor{
+					CustomerId:      customerId,
+					LogType:         logType,
+					AzureResourceId: azureId,
+					Host:            hostCN,
+				},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: secretName,
+			},
+		}
+
+		outputWithTlsSkipVerify = loggingv1.OutputSpec{
+			Type: loggingv1.OutputTypeAzureMonitor,
+			Name: outputName,
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				AzureMonitor: &loggingv1.AzureMonitor{
+					CustomerId: customerId,
+					LogType:    logType,
+				},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: secretName,
+			},
+			TLS: &loggingv1.OutputTLSSpec{
+				InsecureSkipVerify: true,
+			},
+		}
+
+		secrets = map[string]*corev1.Secret{
+			secretName: {
+				Data: map[string][]byte{
+					SharedKey: []byte(sharedKey),
+				},
+			},
+			secretTlsName: {
+				Data: map[string][]byte{
+					SharedKey:          []byte(sharedKey),
+					ClientPrivateKey:   []byte("junk"),
+					ClientCertKey:      []byte("junk"),
+					TrustedCABundleKey: []byte("junk"),
+					Passphrase:         []byte("foo"),
+				},
+			},
+		}
+	)
+
+	DescribeTable("should generate valid config", func(outputSpec loggingv1.OutputSpec, secret, expValue string) {
```

Comment:
Refactoring also introduced some Expectations which make setup of these unit tests easier.  No need to create a generator

Reply:
done